server/services/area.service.js

```diff
@@ -4,24 +4,33 @@ import httpStatus from '../helpers/httpStatus';
 
 export const getAreaById = async (id) => Area.findById(id).select();
 
+export const getAreaByAreaName = async (areaName, stateName) =>
+  Area.find({ area: areaName.toLowerCase(), state: stateName.toLowerCase() }).select();
+
 export const addArea = async (area) => {
+  const areaExists = await getAreaByAreaName(area.area, area.state);
+
+  if (areaExists.length > 0) {
+    throw new ErrorHandler(httpStatus.PRECONDITION_FAILED, 'Area already exists');
+  }
   try {
-    const addedArea = await new Area(area).save();
+    const addedArea = await new Area({
+      ...area,
+      area: area.area.toLowerCase(),
+      state: area.state.toLowerCase(),
+    }).save();
     return addedArea;
   } catch (error) {
     throw new ErrorHandler(httpStatus.BAD_REQUEST, 'Error adding area', error);
   }
 };
 
-export const getStateAndArea = async (state) => {
-  if (!state) {
-    const states = await Area.distinct('state');
-    return { states };
-  }
-  const areas = await Area.find({ state });
-  const area = [];
-  areas.forEach((a) => {
-    area.push({ [a.area]: a._id });
-  });
-  return { areas: area };
+export const getState = async () => {
```

Comment:
One last thing please, to make this method semantically correct, let's update it to the plural form; getStates not getState. 

```suggestion
export const getStates = async () => {
```